### ISSUE TITLE
why fontTools.feaLib.error.FeatureLibError features.fea:561:8: " dflt" is not a valid script tag; use "DFLT" instead ?

### ISSUE THREAD

**Issue by marekjez86** [googlei18n/fontmake#203]
gNotoNastaliqUrdu.glyphs was produced by opening NotoNastaliqUrdu.TTF in glyphs.app and saving it as gNotoNastaliqUrdu.glyphs

[gNotoNastaliqUrdu.glyphs.zip](https://github.com/googlei18n/fontmake/files/632553/gNotoNastaliqUrdu.glyphs.zip)

I used it as an input to fontmake -g gNotoNastaliqUrdu.glyphs -o variable with the following crash.

While gNotoNastaliqUrdu.glyphs might have other problems I am curious why do we require upper case for DFLT? On https://www.microsoft.com/typography/otfntdev/standot/appen.aspx I saw dflt (all lower case). Shouldn't we also accept dflt here?

Traceback (most recent call last):
  File "/usr/local/bin/fontmake", line 9, in <module>
    load_entry_point('fontmake', 'console_scripts', 'fontmake')()
  File "build/bdist.linux-x86_64/egg/fontmake/__main__.py", line 111, in main
  File "build/bdist.linux-x86_64/egg/fontmake/font_project.py", line 348, in run_from_glyphs
  File "build/bdist.linux-x86_64/egg/fontmake/font_project.py", line 391, in run_from_designspace
  File "build/bdist.linux-x86_64/egg/fontmake/font_project.py", line 456, in run_from_ufos
  File "build/bdist.linux-x86_64/egg/fontmake/font_project.py", line 191, in build_interpolatable_ttfs
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/misc/loggingTools.py", line 372, in wrapper
    return func(*args, **kwds)
  File "build/bdist.linux-x86_64/egg/fontmake/font_project.py", line 264, in save_otfs
  File "build/bdist.linux-x86_64/egg/ufo2ft/__init__.py", line 39, in compileTTF
  File "build/bdist.linux-x86_64/egg/ufo2ft/__init__.py", line 28, in compileOTF
  File "build/bdist.linux-x86_64/egg/ufo2ft/makeotfParts.py", line 41, in compile
  File "build/bdist.linux-x86_64/egg/ufo2ft/makeotfParts.py", line 179, in setupFile_featureTables
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/builder.py", line 25, in addOpenTypeFeaturesFromString
    addOpenTypeFeatures(font, featurefile)
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/builder.py", line 16, in addOpenTypeFeatures
    builder.build()
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/builder.py", line 78, in build
    self.parseTree = Parser(self.file).parse()
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/parser.py", line 48, in parse
    statements.append(self.parse_feature_block_())
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/parser.py", line 1059, in parse_feature_block_
    self.parse_block_(block, vertical, stylisticset, size_feature)
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/parser.py", line 1133, in parse_block_
    statements.append(self.parse_script_())
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/parser.py", line 555, in parse_script_
    location, script = self.cur_token_location_, self.expect_script_tag_()
  File "/usr/local/google/home/marekj/.local/lib/python2.7/site-packages/fontTools/feaLib/parser.py", line 1222, in expect_script_tag_
    self.cur_token_location_)
fontTools.feaLib.error.FeatureLibError: /usr/local/google/home/marekj/rel/noto-source/master_ufo/NotoNastaliqUrdu-Regular.ufo/features.fea:561:8: "
dflt" is not a valid script tag; use "DFLT" instead

**Comment by khaledhosny** [comment]
The lower case `dflt` is a language code, while the upper case `DFLT` is script code. The case usually matters here, though in this particular case it is a matter convention since `DFLT` script is just a syntactic sugar and not actually written to the font as such.

**Comment by brawer** [comment]
According to [section 2.c of the feature file specification](http://www.adobe.com/devnet/opentype/afdko/topic_feature_file_syntax.html#2.c), the compiler would be wrong if it accepted lowercase `dflt` for a language tag.

>  DFLT  # can be used only with the script keyword and as the script value with the languagesystem keyword. 
>  dflt  # can be used only with the language keyword and as the language value with the languagesystem keyword.

Indeed, the [OpenType specification](https://www.microsoft.com/typography/otspec/CHAPTER2.htm) writes:

> If a Script table with the script tag 'DFLT' (default) is present in the ScriptList table, it must have a non-NULL DefaultLangSys and LangSysCount must be equal to 0. The 'DFLT' Script table should be used if there is not an explicit entry for the script being formatted.

Now, the binary font file NotoNastaliqUrdu-Regular.ttf contains a script tag `dflt`:

```xml
      <ScriptRecord index="1">
        <ScriptTag value="dflt"/>
        <Script>
          <DefaultLangSys>
            <ReqFeatureIndex value="65535"/>
            <!-- FeatureCount=6 -->
            <FeatureIndex index="0" value="0"/>
            <!-- ... -->
          </DefaultLangSys>
          <!-- LangSysCount=0 -->
        </Script>
      </ScriptRecord>
```

In other words, the input font is broken and should be fixed. Instead of `dflt`, it should use `DFLT` as the script tag. Filed https://github.com/googlei18n/noto-fonts/issues/805. Closing this bug because the pipeline is correct in rejecting this input.
